zigen-project/zen
# Make session owner zen

## Context

In znr-remote, Session (c++) is handled as shared_ptr, which means owned by multiple objects.
In zen, we have a znr_session object, and the owner of znr_session object was ambiguous. (znr implementation or zen)

This pull request makes the owner of znr_session zen.

## Summary

- [x] znr_session is now created and destroyed by zen (not znr-remote or other znr implementation)
- [x] To elaborate, zna will take ownership.
- [x] https://github.com/zigen-project/zen-remote/pull/26 is properly applied

## How to check behavior

1. start zen
2. start zen-oculus
3. Confirm that a new session started. (see log)
4. terminate zen-oculus
5. start zen-box (sample client)
6. Confirm that the session end. (see log)
